tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/ClientSideEncryptionTestRunner.cs

```diff
@@ -0,0 +1,317 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Tests.Specifications.Runner;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.client_side_encryption
+{
+    // todo: SPEC-1385
+    public class ClientSideEncryptionTestRunner : MongoClientJsonDrivenTestRunnerBase
+    {
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void Run(JsonDrivenTestCase testCase)
+        {
+            var shared = RestoreGuidRepresentations(testCase.Shared);
```

Comment:
Note: it looks like there is a bug in vs + xunit. I will describe it a little bit later.

Reply:
@rstam , sorry I forgot to return to this comment in the previous round.
So, it looks like we have a vs bug here (You can also check whether this situation is reproduced on your machine or not).
1. With the currently added lines (with `RestoreGuidRepresentations`) all tests pass.
2. Without these lines - some tests failed with the error: `Command create failed: Array elements must have bindata type UUID, found 3.` for `.net core` tests.

But when I apply this lifehack for the method `ReadJsonDocument`:
```
       #pragma warning disable
       public static bool debug = true;
       #pragma warning restore

        protected virtual BsonDocument ReadJsonDocument(string path)
        {
            if (debug)
            {
                throw new Exception("test");
            }
            var jsonReaderSettings1 = new JsonReaderSettings
            {
                GuidRepresentation = GuidRepresentation.Unspecified
            };
            using (var stream = Assembly.GetManifestResourceStream(path))
            using (var streamReader = new StreamReader(stream))
            using (var jsonReader = new JsonReader(streamReader, jsonReaderSettings1))
            // other code lines
```

here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/48/files#diff-26821dbb07823b7230d1116ab8335d72R27
I can "fix" the problem. I guess it happens because `vs` cannot make any sort of "caching" since this method is not executed correctly.

To pass tests, I'm setting a breakpoint in the line with throwing exception. And when I run tests, I just manually set `debug=false` before exception has been thrown. It allow `vs` to read all files and pass tests.

I don't know how this situation can be fixed in another way (not like I implemented via `RestoreGuidRepresentations`), but without this method, `net core` tests have wrong `guid representation` values.

Reply:
NOTE: this error is not reproduced via: `.\build.ps1 -target test`

Reply:
Repeat: if the JSON files are read correctly there should be no need to have or call a method like `RestoreGuidRepresentations`.

Reply:
I can confirm that the tests pass with netfw an IDE (Rider in my case). The tests pass for both netcore and netfw via the cli (`dotnet test`). Using Dima's hack, I was able to get tests that were previously failing for .NET Core to pass in IDE (although my test runner hung and I only saw 25 tests go green whereas all of them had failed before)

Reply:
@DmitryLukyanov Does it work in an IDE If we were to use to the older [JsonDrivenTestCaseFactory](https://github.com/vincentkam/mongo-csharp-driver/blob/7cd604ab4fee1c3a02fd56e7b3f99215055785f4/tests/MongoDB.Driver.Tests/Specifications/retryable-reads/RetryableReadsTestRunner.cs#L412)? If so, in the interest of time, perhaps we can use the old factory and we can create create a ticket to track the current issue with the `EmbededJsonResourceTestRunner`?